internal/providers/system/system.go

```diff
@@ -30,12 +32,25 @@ import (
 )
 
 const (
-	baseFilename = "base.ign"
 	userFilename = "user.ign"
 )
 
-func FetchBaseConfig(logger *log.Logger) (types.Config, report.Report, error) {
-	return fetchConfig(logger, baseFilename)
+func FetchBaseConfig(logger *log.Logger, platformName string) (types.Config, report.Report, error) {
+	var fullBaseConfig types.Config
+	baseDConfig, baseDReport, err := fetchBaseConfig(logger, "base.d")
+	if err != nil {
+		logger.Warning("no config at base.d: %v", err)
+	}
+	fullBaseConfig = latest.Merge(fullBaseConfig, baseDConfig)
```

Comment:
Actually, we don't need to merge here; we can just fetch directly into `fullBaseConfig`.